src/app/shared/skill-select-input/skill-select-input.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { MatAutocomplete, MatAutocompleteSelectedEvent, MatChipInputEvent } from '@angular/material';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Observable, of } from 'rxjs';
+import { Skill } from '../../skills/skill';
+import { COMMA, ENTER } from '@angular/cdk/keycodes';
+import { SkillsService } from '../../skills/skills.service';
+import { map, startWith } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-skill-select-input',
+  templateUrl: './skill-select-input.component.html',
+  styleUrls: ['./skill-select-input.component.scss']
+})
+export class SkillSelectInputComponent implements OnInit {
+
+  @Input() parentForm: FormGroup;
+
+  @ViewChild('skillsAutoComplete') skillsMatAutocomplete: MatAutocomplete;
+  @ViewChild('skillInput') skillAutocompleteInput: ElementRef<HTMLInputElement>;
+
+  errorMessage: string = null;
+  skills$: Observable<Skill[]> = of([]);
+  allAvailableSkills: Skill[];
+
+  elemSelectable = true;
```

Comment:
Wouldn't it be better to use hard-coded constants `true / false` in the component template instead? We could avoid defining of the additional component fields which are used only in the component template. I can imagine the case when we will need to access these fields in unit-tests (for example to test component behaviour like if an element can be removed or selected). But for now we don't have such unit-tests. I consider these fields to be just an additional state of the component. I think we could replace the fields with the boolean constants in the template. What do you think?

Reply:
thanks. done